packages/headless/src/controllers/commerce/search-box/headless-search-box.ts

```diff
@@ -0,0 +1,188 @@
+  clearQuerySuggest,
+  registerQuerySuggest,
+  selectQuerySuggestion,
+} from '../../../features/query-suggest/query-suggest-actions';
+import {querySuggestReducer as querySuggest} from '../../../features/query-suggest/query-suggest-slice';
+import {
+  CommerceQuerySection,
+  CommerceSearchSection,
+  ConfigurationSection,
+  QuerySetSection,
+  QuerySuggestionSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  Delimiters,
+  SuggestionHighlightingOptions,
+} from '../../../utils/highlight';
+import {randomID} from '../../../utils/utils';
+import {validateOptions} from '../../../utils/validate-payload';
+import {buildController} from '../../controller/headless-controller';
+import {
+  getSuggestions,
+  SearchBoxState,
+  SearchBox as CoreSearchBox,
+  Suggestion,
+} from '../../core/search-box/headless-core-search-box';
+import {
+  defaultSearchBoxOptions,
+  SearchBoxOptions,
+  searchBoxOptionsSchema,
+} from './headless-search-box-options';
+
+export type {
+  SearchBoxOptions,
+  SearchBoxState,
+  SuggestionHighlightingOptions,
+  Suggestion,
+  Delimiters,
+};
+
+export interface SearchBoxProps {
+  /**
+   * The `SearchBox` controller options.
+   */
+  options?: SearchBoxOptions;
+}
+
+/**
+ * The `SearchBox` headless controller offers a high-level interface for designing a common search box UI controller
+ * with [highlighting for query suggestions](https://docs.coveo.com/en/headless/latest/usage/highlighting/).
+ */
+export type SearchBox = Omit<CoreSearchBox, 'submit'> & {
+  /**
+   * Deselects all facets and triggers a commerce search query.
+   */
+  submit(): void;
+};
+
+/**
+ * Creates a `SearchBox` controller instance.
+ *
+ * @param engine - The commerce headless engine.
+ * @param props - The configurable `SearchBox` properties.
+ * @returns A `SearchBox` controller instance.
+ */
+export function buildSearchBox(
+  engine: CommerceEngine,
+  props: SearchBoxProps = {}
+): SearchBox {
+  if (!loadSearchBoxReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  const id = props.options?.id || randomID('search_box');
+  const options: Required<SearchBoxOptions> = {
+    id,
+    highlightOptions: {...props.options?.highlightOptions},
+    ...defaultSearchBoxOptions,
+    ...props.options,
+  };
+
+  validateOptions(engine, searchBoxOptionsSchema, options, 'buildSearchBox');
+  dispatch(
+    registerQuerySetQuery({id, query: engine.state.commerceQuery.query ?? ''})
+  );
+  dispatch(registerQuerySuggest({id}));
+
+  const getValue = () => engine.state.querySet[options.id];
+
+  const performSearch = async () => {
+    if (options.clearFilters) {
+      dispatch(deselectAllBreadcrumbs());
+    }
+
+    dispatch(updateFacetAutoSelection({allow: true}));
+    dispatch(updateQuery({query: getValue()}));
+    dispatch(selectPage(1));
```

Comment:
Maybe we should instead dispatch an action when we start performing the search request, and instead of resetting pagination here, we'd add a case in the pagination reducer.

Cause right now we're dispatching a pagination action without loading the pagination reducer, which feels kind of strange to me.

Reply:
Done in https://github.com/coveo/ui-kit/pull/3461/commits/44293853f8c134a03c6136858d077ccb9a2b475d